app/client/src/features/accounts/ResetPasswordForm.tsx

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable react/jsx-curly-newline */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { Button, IconButton, InputAdornment, Grid } from '@material-ui/core';
+import Visibility from '@material-ui/icons/Visibility';
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import { makeStyles } from '@material-ui/core/styles';
+import { useMutation } from 'react-relay';
+import { Form } from '@local/components/Form';
+import { FormContent } from '@local/components/FormContent';
+import { TextField } from '@local/components/TextField';
+import { useSnack, useForm } from '@local/features/core';
+import { useRouter } from 'next/router';
+import { ResetPasswordRequestMutation } from '@local/__generated__/ResetPasswordRequestMutation.graphql';
+import { RESET_PASSWORD_FORM_MUTATION } from './ResetPasswordRequest';
+
+interface Props {
+    onSuccess: () => void;
+    onFailure: () => void;
+}
+
+const initialState = {
+    newPassword: '',
+    confirmNewPassword: '',
+};
+
+export type TResetPasswordForm = typeof initialState;
+
+const useStyles = makeStyles((theme) => ({
+    form: {
+        margin: theme.spacing(0, 1, 0, 1),
+    },
+}));
+
+export function ResetPasswordForm({ onSuccess, onFailure }: Props) {
+    const [isPassVisible, setIsPassVisible] = React.useState(false);
+
+    // form state hooks
+    const [form, errors, handleSubmit, handleChange] = useForm(initialState);
+    const [commit] = useMutation<ResetPasswordRequestMutation>(RESET_PASSWORD_FORM_MUTATION);
+
+    // user feedback
+    const { displaySnack } = useSnack();
+
+    // styling hook
+    const classes = useStyles();
+
+    const router = useRouter();
+
+    function handleCommit(submittedForm: TResetPasswordForm) {
+        commit({
+            variables: { 
+                input: {
+                    token: router.query.token ? String(router.query.token) : '',
+                    ...submittedForm
+                } 
+            },
+            onCompleted({ resetPassword }) {
+                if (resetPassword.isError) {
+                    if (resetPassword.message === 'Invalid token.') {
+                        displaySnack('Error: Invalid token. Redirecting back to homepage.');
+                        router.push('/')
+                    }
+                    else {
+                        displaySnack(resetPassword.message);
+                    }
+                } else {
+                    displaySnack('Password changed successfully!');
+                    if (onSuccess) onSuccess();
+                }
```

Comment:
I don't think prettier was run on these files, the formatting seems off.  Make sure that you have the `prettier` vscode extension installed to autoformat your code.